src/image.rs

```diff
@@ -555,10 +562,10 @@ pub trait ImageDecoder<'a>: Sized {
     /// This is the size of the buffer that must be passed to `read_image` or
     /// `read_image_with_progress`. The returned value may exceed usize::MAX, in
     /// which case it isn't actually possible to construct a buffer to decode all the image data
-    /// into.
+    /// into. If, however, the size does not fit in a u64 then u64::MAX is returned.
     fn total_bytes(&self) -> u64 {
         let dimensions = self.dimensions();
-        u64::from(dimensions.0) * u64::from(dimensions.1) * u64::from(self.color_type().bytes_per_pixel())
+        (u64::from(dimensions.0) * u64::from(dimensions.1)).saturating_mul(u64::from(self.color_type().bytes_per_pixel()))
```

Comment:
This suffers from small readability problem now, rather split up in two lines:
```suggestion
        let pixel_count = u64::from(dimensions.0) *u64::from(dimensions.1);
        let bpp = u64::from(self.color_type().bytes_per_pixel());
        pixel_count.saturating_mul(bpp)
```